File: plugin/graph/src/main/groovy/net/twisterrob/gradle/graph/vis/TaskVisualizer.java
package net.twisterrob.gradle.graph.vis;

import java.util.Map;

import org.gradle.api.*;

import net.twisterrob.gradle.graph.tasks.*;

public interface TaskVisualizer {
	void showUI(Project project);
	void initModel(Map<Task, TaskData> graph);
	void update(Task task, TaskResult result);
	/**
	 * Responsible for
	 *
	 * Closing the UI:
	 * <li>hiding if the UI is reusable
	 * <li>disposing of it if it needs to be recreated next time
	 *
	 * The cache received in the constructor should be also closed, always, even if the UI failed.
	 */
	void closeUI();
}

File: graph/src/main/kotlin/net/twisterrob/gradle/graph/vis/TaskVisualizer.kt
package net.twisterrob.gradle.graph.vis

import net.twisterrob.gradle.graph.tasks.TaskData
import net.twisterrob.gradle.graph.tasks.TaskResult
import org.gradle.api.Task
import org.gradle.api.initialization.Settings

interface TaskVisualizer {

	fun showUI(settings: Settings)
	fun initModel(graph: Map<Task, TaskData>)
	fun update(task: Task, result: TaskResult)

	/**
	 * Responsible for
	 *
	 * Closing the UI:
	 *  * hiding if the UI is reusable
	 *  * disposing of it if it needs to be recreated next time
	 *
	 * The cache received in the constructor should be also closed, always, even if the UI failed.
	 */
	fun closeUI()
}
